tests/models/opt/test_modeling_opt.py

```diff
@@ -281,7 +281,7 @@ def test_inference_no_head(self):
         attention_mask = input_ids.ne(model.config.pad_token_id)
         with torch.no_grad():
             output = model(input_ids=input_ids, attention_mask=attention_mask).last_hidden_state
-        expected_shape = torch.Size((1, 11, 1024))
+        expected_shape = torch.Size((1, 11, 512))
```

Comment:
I don't understand very well why it is expected to be `512`.

Fomr [config](https://huggingface.co/facebook/opt-350m/blob/main/config.json), I see

```
 "hidden_size": 1024,
 "word_embed_proj_dim": 512,
```
So any special reason for `last_hidden_state` to be `512`? It looks unusual if it is intended to be so - therefore a comment would be very helpuful in this casee.

Reply:
If my understanding is correct, for some particular cases in OPT - the word embeddings have a different size than the dimension of the internal hidden states. That is why in the config file you have the `word_embed_proj_dim` arg. 
So basically the hidden states that are computed [here](https://github.com/huggingface/transformers/blob/85fc4559722138416c477b08bc3e8e0e07fce5af/src/transformers/models/opt/modeling_opt.py#L741) including the last hidden states will have a dimension of  `word_embed_proj_dim`. The conversion from `hidden_size` to `word_embed_proj_dim` is ensured by the `self.project_out` and `self.project_in` modules

Reply:
Thank you @younesbelkada , I understand very well now 😄. I still think it would be a very good idea to add a (short) comment about what you mentioned above.

Regarding this, I personally think it would be much better to have something like:

```
@dataclass
class OPTModelOutput(ModelOutput):

    last_hidden_state  = None
    projected_word_embedding = None  # or whatever better name
   ... (remaining fields )
```
I will leave @patrickvonplaten and @LysandreJik to judge though, they have much more experience than me.

Reply:
Well, I still expect a comment to be included in the code 😢